php/commands/help.php

```diff
@@ -37,15 +37,9 @@ function __invoke( $args, $assoc_args ) {
 			self::show_help( $command );
 			exit;
 		}
-
-		// WordPress is already loaded, so there's no chance we'll find the command
-		if ( function_exists( 'add_filter' ) ) {
-			$command_string = implode( ' ', $args );
-			\WP_CLI::error( sprintf( "'%s' is not a registered wp command.", $command_string ) );
-		}
```

Comment:
I've changed it now to use `Runner::find_command_to_help()` to address https://github.com/wp-cli/wp-cli/pull/4266#discussion_r131228598 fully, and removed `find_command_to_help()` (previously `Help_Command::find_subcommand()`) altogether - I'll expand below as the review commenting is getting a bit messy.